external/fv3fit/fv3fit/emulation/microphysics/models.py

```diff
@@ -0,0 +1,206 @@
+        ]
+
+        return inputs
+
+    def _get_direct_outputs(self, sample_out, net_output):
+
+        outputs = []
+
+        for name, sample in zip(self.direct_out_variables, sample_out):
+            out_ = FieldOutput(
+                sample,
+                denormalize=self.normalize_key,
+                name=name,
+                enforce_positive=True,
+            )(net_output)
+            outputs.append(out_)
+
+        return outputs
+
+    def _get_residual_outputs(self, sample_out, net_output, residual_to_input_map):
+
+        outputs = []
+        tendencies = []
+
+        for (name, in_state), sample in zip(residual_to_input_map.items(), sample_out):
+            # incremented state field output
+            res_out = IncrementedFieldOutput(
+                sample,
+                self.timestep_increment_sec,
+                denormalize=self.normalize_key,
+                name=name,
+                enforce_positive=True,
+            )
+
+            out_ = res_out(in_state, net_output)
+            outputs.append(out_)
+
+            if name in self.tendency_outputs:
+                tend_name = self.tendency_outputs[name]
+                tendency = res_out.get_tendency_output(net_output)
+                renamed = tf.keras.layers.Lambda(lambda x: x, name=tend_name)(tendency)
+                tendencies.append(renamed)
+
+        return outputs + tendencies
+
+    def build(
+        self,
+        sample_in: Sequence[tf.Tensor],
+        sample_direct_out: Sequence[tf.Tensor] = None,
+        sample_residual_out: Sequence[tf.Tensor] = None,
+    ):
+        """
+        Build model described by the configuration
+
+        Args:
+            sample_in: Sample input tensors for determining layer shapes and
+                fitting normalization layers if specifies
+            sample_direct_out: Sample of direct output field tensors used to
+                fit denormalization layers (if specified) and shape
+            sample_residual_out: Sample of residual output field tensors used to
+                fit denormalization layers (if specified) and shape.  Note: the
+                samples should be in tendency form to produce proper denormalization
+        """
+
+        if sample_direct_out is None:
+            sample_direct_out = []
+
+        if sample_residual_out is None:
+            sample_residual_out = []
+
+        inputs = [
+            tf.keras.layers.Input(sample.shape[-1], name=name)
+            for name, sample in zip(self.input_variables, sample_in)
+        ]
+        residual_map = {
+            resid_name: inputs[self.input_variables.index(input_name)]
+            for resid_name, input_name in self.residual_out_variables.items()
+        }
+        processed = self._get_processed_inputs(sample_in, inputs)
+        combine_layer = get_combine_from_arch_key(self.architecture.name)
+        combined = combine_layer(processed)
+        arch_layer = self.architecture.build
```

Comment:
```suggestion
        arch_layer = self.architecture.build()
```

Reply:
This used to be a `@property` field of the class as `instance`.  I forgot to switch it over.  I'll make this adjustment in a single commit.